tests/testsuite/alt_registry.rs

```diff
@@ -416,7 +416,16 @@ or use environment variable CARGO_REGISTRIES_ALTERNATIVE_TOKEN",
 #[cargo_test]
 fn publish_to_alt_registry() {
     registry::alt_init();
-    let p = project().file("src/main.rs", "fn main() {}").build();
+    let p = project()
+        .file("src/main.rs", "fn main() {}")
+        .file(
+            ".cargo/config",
+            "
+                [publish]
+                timeout = 0
```

Comment:
Before, our goal was to allow the tests to run as-close to production as possible which would make the blocking code decide not to block until timeout, rather than just disabling the timeout.

Would it be possible for us to get that working again?

Reply:
Note this part in the issue

> Change these tests so that they use with_stderr to validate their output (which would have caught this regression).

`with_stderr` can help us detect when we are hitting the timeout

Reply:
#11217 (a subset of #11062) provides some background, including an exploration of alternatives.  The short version is I changed the tests that I felt didn't care what went into the registry but left the ones that we wanted to validate what was published to the registry.  Note that it isn't just validating the `.crate` file but the API call itself.  I had originally discarded it as not being worth the time but if it works out for you, sure.